protocols/medco/private_aggregate_simul.go

```diff
@@ -0,0 +1,145 @@
+	testCVMap := make(map[libmedco.GroupingKey]libmedco.CipherVector)
+
+	for i := 0; i < numberGroups; i++ {
+		newGrpattr := grpattr
+		(libmedco.DeterministCipherText(newGrpattr).Point).Add(libmedco.DeterministCipherText(newGrpattr).Point,
+			libmedco.DeterministCipherText(newGrpattr).Point)
+		groupAttributes := libmedco.GroupingAttributes{grpattr, newGrpattr}
+
+		grpattr = newGrpattr
+
+		var tab []int64
+		for i := 0; i < numberAttributes; i++ {
+			if i == 0 {
+				tab = []int64{1}
+			} else {
+				tab = append(tab, 1)
+			}
+		}
+		//round := monitor.NewTimeMeasure("MEDCO_ENCRYPTION")
+		cipherVect := *libmedco.EncryptIntVector(clientPublic, tab)
+		//round.Record()
+
+		testGAMap[groupAttributes.Key()] = groupAttributes
+		testCVMap[groupAttributes.Key()] = cipherVect
+	}
+	return testGAMap, testCVMap
+}
+
+func init() {
+	sda.SimulationRegister("PrivateAggregate", NewPrivateAggregateSimulation)
+	sda.ProtocolRegisterName("PrivateAggregateSimul", NewAggregationProtocolSimul)
+}
+
+//PrivateAggregateSimulation contains simulation tree
+type PrivateAggregateSimulation struct {
+	sda.SimulationBFTree
+}
+
+//NewPrivateAggregateSimulation simultaion constructor
+func NewPrivateAggregateSimulation(config string) (sda.Simulation, error) {
+	sim := &PrivateAggregateSimulation{}
+	_, err := toml.Decode(config, sim)
+	if err != nil {
+		return nil, err
+	}
+	return sim, nil
+}
+
+//Setup initializes the servers tree
+func (sim *PrivateAggregateSimulation) Setup(dir string, hosts []string) (*sda.SimulationConfig, error) {
+	sc := &sda.SimulationConfig{}
+	sim.CreateRoster(sc, hosts, 20)
+	err := sim.CreateTree(sc)
+
+	if err != nil {
+		return nil, err
+	}
+
+	log.Lvl1("Setup done")
+
+	return sc, nil
+
+}
+
+//Run starts the simulation of the protocol and measures its runtime
+func (sim *PrivateAggregateSimulation) Run(config *sda.SimulationConfig) error {
+	for round := 0; round < sim.Rounds; round++ {
+		log.Lvl1("Starting round", round)
+		rooti, err := config.Overlay.CreateProtocolSDA(config.Tree, "PrivateAggregateSimul")
+		if err != nil {
+			return err
+		}
+
+		root := rooti.(*PrivateAggregateProtocol)
+
+		//time measurement
+		round := monitor.NewTimeMeasure("MEDCO_PROTOCOL")
+
+		root.StartProtocol()
+
+		<-root.ProtocolInstance().(*PrivateAggregateProtocol).FeedbackChannel
+		round.Record()
+	}
+
+	return nil
+}
+
+//NewAggregationProtocolSimul default constructor used by all nodes to init their parameters
+func NewAggregationProtocolSimul(tni *sda.TreeNodeInstance) (sda.ProtocolInstance, error) {
+	protocol, err := NewPrivateAggregate(tni)
+	pap := protocol.(*PrivateAggregateProtocol)
+
+	groupMap := make(map[libmedco.GroupingKey]libmedco.GroupingAttributes)
+	attribMap := make(map[libmedco.GroupingKey]libmedco.CipherVector)
+
+	switch tni.Index() {
+	//if want to study special cases********************************************************************************
```

Comment:
Unused code? or should it be made optional in some way? Or only being implemented in the tests?